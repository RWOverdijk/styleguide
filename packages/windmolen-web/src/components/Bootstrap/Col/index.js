import React from 'react';
import styled from 'styled-components';
import { Col as BootstrapCol } from 'styled-bootstrap-grid';

const Col = styled(BootstrapCol)``;

export default Col;
