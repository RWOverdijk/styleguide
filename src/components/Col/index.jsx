import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './style.scss';

class Col extends PureComponent {
  getClass() {
    let className = 'col-';

    if (typeof this.props.sm === 'string') {
      className += `sm-${this.props.sm}`;
    } else if (typeof this.props.md === 'string') {
      className += `md-${this.props.md}`;
    } else if (typeof this.props.lg === 'string') {
      className += `lg-${this.props.lg}`;
    }

    return className;
  }

  render() {
    return <div className={this.getClass()}>{this.props.children}</div>;
  }
}

Col.defaultProps = {
  sm: '',
  md: '',
  lg: '',
  children: '',
};

Col.propTypes = {
  sm: PropTypes.string,
  md: PropTypes.string,
  lg: PropTypes.string,
  children: PropTypes.oneOf([
    PropTypes.string,
    PropTypes.element,
  ]),
};

export default Col;
