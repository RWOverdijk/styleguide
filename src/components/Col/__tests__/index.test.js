import React from 'react';
import renderer from 'react-test-renderer';
import Col from '../';

describe("<Col>", () => {
  it("renders Col with xs size 5", () => {
    const tree = renderer.create(
      <Col xs={5}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Col with sm size 5", () => {
    const tree = renderer.create(
      <Col sm={5}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Col with md size 5", () => {
    const tree = renderer.create(
      <Col md={5}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Col with lg size 5", () => {
    const tree = renderer.create(
      <Col lg={5}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Col with xs, sm, sm and lg size 3", () => {
    const tree = renderer.create(
      <Col xs={3} sm={3} md={3} lg={3}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Col with offset-xs size 5", () => {
    const tree = renderer.create(
      <Col xsOffset={5} xs={5}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Col with offset-sm size 5", () => {
    const tree = renderer.create(
      <Col smOffset={5} sm={5}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Col with offset-md size 5", () => {
    const tree = renderer.create(
      <Col mdOffset={5} md={5}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Col with offset-lg size 5", () => {
    const tree = renderer.create(
      <Col lgOffset={5} lg={5}>My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Col xs={5} className="my-class">My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Col xs={5} data-id="0">My Col</Col>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
