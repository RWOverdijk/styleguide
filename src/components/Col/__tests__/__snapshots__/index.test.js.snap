// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col> renders Col with lg size 5 1`] = `
<div
  className="col-lg-5"
>
  My Col
</div>
`;

exports[`<Col> renders Col with md size 5 1`] = `
<div
  className="col-md-5"
>
  My Col
</div>
`;

exports[`<Col> renders Col with offset-lg size 5 1`] = `
<div
  className="col-lg-5 col-lg-offset-5"
>
  My Col
</div>
`;

exports[`<Col> renders Col with offset-md size 5 1`] = `
<div
  className="col-md-5 col-md-offset-5"
>
  My Col
</div>
`;

exports[`<Col> renders Col with offset-sm size 5 1`] = `
<div
  className="col-sm-5 col-sm-offset-5"
>
  My Col
</div>
`;

exports[`<Col> renders Col with offset-xs size 5 1`] = `
<div
  className="col-xs-5 col-xs-offset-5"
>
  My Col
</div>
`;

exports[`<Col> renders Col with sm size 5 1`] = `
<div
  className="col-sm-5"
>
  My Col
</div>
`;

exports[`<Col> renders Col with xs size 5 1`] = `
<div
  className="col-xs-5"
>
  My Col
</div>
`;

exports[`<Col> renders Col with xs, sm, sm and lg size 3 1`] = `
<div
  className="col-xs-3 col-sm-3 col-md-3 col-lg-3"
>
  My Col
</div>
`;

exports[`<Col> renders with specific attributes 1`] = `
<div
  className="col-xs-5"
  data-id="0"
>
  My Col
</div>
`;

exports[`<Col> renders with specific className 1`] = `
<div
  className="col-xs-5 my-class"
>
  My Col
</div>
`;
